software/o_c_REV/APP_ENVGEN.ino

```diff
@@ -192,9 +198,19 @@ public:
 
     OC::DigitalInput trigger_input = get_trigger_input();
     uint8_t gate_state = 0;
-    if (triggers & DIGITAL_INPUT_MASK(trigger_input))
-      gate_state |= peaks::CONTROL_GATE_RISING;
 
+    bool triggered = triggered_;
+    if (triggers & DIGITAL_INPUT_MASK(trigger_input)) {
+      sinceTrigger_ = 0;
+      triggered=true;
+    }
+    uint16_t trigger_delay = get_trigger_delay();
+    if ((!trigger_delay && triggered) || (triggered && (sinceTrigger_ >= trigger_delay))) {
```

Comment:
A lot of Arduino libraries/code isn't safe to use in ISRs so I tend to stay away from it. If that's from Paul it might be ok.